### PR TITLE
Clean up the reset code for button's rounded corners in button group and input button

### PR DESCRIPTION
**Pull request by ysds**
A `.dropdown-toggle` selector is used to reset a button's rounded corner in button group and input button. But should not use a `dropdown-toggle` selector to reset it, because `dropdown-toggle` does not have rounded corners by default.

This PR including the follwings:
- Use `last-of-type` selector instead of `.dropdown-toggle`
- Remove `border-raduis :0` because it is enough by each reset of the left and right side

I hope that this PR will help improve code. Thanks.

### CODE REVIEW

**Comment by andresgalante** [comment]
Hi @ysds Thanks for the PR. I found that there is still a rounded corner on the last btn before the drop menu. Look a the first example under "nesting":

screen shot 2017-11-28 at 8 57 34 am

It's really challenging to remove or simplify code, thanks a lot for giving it a try.

**Comment by ysds** [comment]
Hi @andresgalante Thank you for your review! and sorry for my insufficient check. Using `last-of-type` instead of `.dropdown-toggle` seems does not seem to be appropriate.
I modified and restored some selector, please double check it.

**Review by andresgalante — APPROVED** [review]
Thanks a lot @ysds, I have tested and review the code and I it looks ok to me.

But in case I am missing something I'll assign this PR to @mdo to take another look. 

The comment on line 57 is confusing to me, maybe we can improve the language.

File: scss/_button-group.scss [hunk 1]
```diff
@@ -44,46 +44,25 @@
...
-  > .btn:last-child:not(:first-child),
-  > .dropdown-toggle:not(:first-child) {
-    @include border-left-radius(0);
   }
 
-  // Custom edits for including btn-groups within btn-groups (useful for including
+  // Alignment for including btn-groups within btn-groups (useful for including
   // dropdown buttons within a btn-group)
   > .btn-group {
     float: left;
```
Comment (andresgalante) [inline scss/_button-group.scss]: Do we actually need this float here? What is it for?

**Comment by ysds** [comment]
@andresgalante 

I updated the comment more concisely.

> Do we actually need this float here? What is it for?

I think so too. it will not be needed. and there seems to be some other code that can be deleted/clean-up.

e.g: L47-
```scss
.btn-group {
  > .btn:first-child {
    margin-left: 0;
  }
```
and more. I can additional cleanup with another PR if this PR is approved. Thanks.

**Comment by mdo** [comment]
Changes to the input group here will need to be removed once #25020 lands.

**Comment by ysds** [comment]
Please follow #25073 😄

**Pull request closed without merging**